### ISSUE TITLE
Getting an ERROR_DEVICE_LOST

### ISSUE THREAD

**Issue by gzp-crey** [gfx-rs/wgpu-rs#420]
I'm getting an ERROR_DEVICE_LOST for a test project that was working a few days (at most 2 weeks) ago.
wgpu-rs: main branch @revision 0c7c3afebb12bf411d6964887b85e5fd152057f5

See the attached trace.
It has nothing special in it, just renders a simple mesh with texture using a very simple shader. In each frame two encoder is created (and submitted separately, one to update the uniform and another for the rendering.

Any hint would be welcome.

[wgpu_trace.zip](https://github.com/gfx-rs/wgpu-rs/files/4857437/wgpu_trace.zip)

**Comment by gzp-crey** [comment]
Could be a duplicate of #392 ?

**Comment by kvark** [comment]
@gzp-crey thank you for filing! It's so much easier to look at stuff when there is an API trace attached ❤️ 
I ran it through the player and got an assertion:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `0`: Buffer write size 18 must be a multiple of 4', wgpu-core/src/device/queue.rs:187:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
(had to fix the message a tiny bit)

So my reading is that we are failing to copy the contents over from the staging area, because the buffer size is not aligned to 4. This is not your problem, we should be handling this internally (looking now).

**Comment by gzp-crey** [comment]
I've tried it with an index buffer size of 36 (instead of 18)  and the crash is still present. I see no other buffer that has a size not aligned to 4.

I don't know if it is relevant, but if I remove the use of the UniformBuffer (the projection matrix), there is no crash and the scene is rendered correctly with sampler and texture, (the size of the uniform buffer is 64).

**Comment by kvark** [comment]
Thanks for checking this! Do you have Vulkan validation layers installed and logging enabled? Anything interesting reported there?

**Comment by kvark** [comment]
Also, would you mind sharing an API trace with the fixed index buffer size?

**Comment by gzp-crey** [comment]
I've installed the validation layer and it turned out that the shader and descriptor were mismatching.
In shader I had set=0,bind=0 for texture, set=0,bind=1 for sampler and set=1,bind=0 for uniform.
In pipeline there was a single binding group with 3 binding. 
The strange thing is that it was working on one computer and failed on another but I have no idea how could it resolve the shader parameters for the working driver. That was a mere accident it was working at all.

Thanks for the help and as usual it was an user error :). Unfortunately I have to wait until Monday to validate and test my fix on the other PC.
